File: xamarin/Framework/LiferayScreens.Android/Additions/ImageGallery/ImageGalleryListener.cs
using Android.Runtime;
using Com.Liferay.Mobile.Screens.Base.Interactor.Listener;
using Com.Liferay.Mobile.Screens.Imagegallery.Model;
using Com.Liferay.Mobile.Screens.Base.List;
using System;

namespace Com.Liferay.Mobile.Screens.Imagegallery
{
    [Register("com/liferay/mobile/screens/imagegallery/ImageGalleryListener", "", "Com.Liferay.Mobile.Screens.Imagegallery.IImageGalleryListenerInvoker")]
    public interface IImageGalleryListener : IBaseListListener, IBaseCacheListener, IJavaObject, IDisposable
    {
        //
        // Methods
        //
        [Register("onImageEntryDeleted", "(J)V", "GetOnImageEntryDeleted_JHandler:Com.Liferay.Mobile.Screens.Imagegallery.IImageGalleryListenerInvoker, LiferayScreens_Android")]
        void OnImageEntryDeleted(long p0);

        [Register("onImageUploadEnd", "(Lcom/liferay/mobile/screens/imagegallery/model/ImageEntry;)V", "GetOnImageUploadEnd_Lcom_liferay_mobile_screens_imagegallery_model_ImageEntry_Handler:Com.Liferay.Mobile.Screens.Imagegallery.IImageGalleryListenerInvoker, LiferayScreens_Android")]
        void OnImageUploadEnd(ImageEntry p0);

        [Register("onImageUploadProgress", "(II)V", "GetOnImageUploadProgress_IIHandler:Com.Liferay.Mobile.Screens.Imagegallery.IImageGalleryListenerInvoker, LiferayScreens_Android")]
        void OnImageUploadProgress(int p0, int p1);

        [Register("onImageUploadStarted", "(Landroid/net/Uri;Ljava/lang/String;Ljava/lang/String;Ljava/lang/String;)V", "GetOnImageUploadStarted_Landroid_net_Uri_Ljava_lang_String_Ljava_lang_String_Ljava_lang_String_Handler:Com.Liferay.Mobile.Screens.Imagegallery.IImageGalleryListenerInvoker, LiferayScreens_Android")]
        void OnImageUploadStarted(Java.Net.URI p0, string p1, string p2, string p3);

        [Register("provideImageUploadDetailView", "()I", "GetProvideImageUploadDetailViewHandler:Com.Liferay.Mobile.Screens.Imagegallery.IImageGalleryListenerInvoker, LiferayScreens_Android")]
        int ProvideImageUploadDetailView();

        [Register("showUploadImageView", "(Ljava/lang/String;Landroid/net/Uri;I)Z", "GetShowUploadImageView_Ljava_lang_String_Landroid_net_Uri_IHandler:Com.Liferay.Mobile.Screens.Imagegallery.IImageGalleryListenerInvoker, LiferayScreens_Android")]
        bool ShowUploadImageView(string p0, Java.Net.URI p1, int p2);
    }
}

namespace Com.Liferay.Mobile.Screens.Imagegallery.Model
{
    public partial class ImageEntry { }
}

File: android/library/src/debug/java/com/liferay/mobile/screens/imagegallery/ImageGalleryListener.java
/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 * <p/>
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 * <p/>
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

package com.liferay.mobile.screens.imagegallery;

import android.net.Uri;
import androidx.annotation.LayoutRes;
import com.liferay.mobile.screens.base.list.BaseListListener;
import com.liferay.mobile.screens.imagegallery.model.ImageEntry;
import com.liferay.mobile.screens.viewsets.defaultviews.imagegallery.DefaultUploadDetailView;

/**
 * @author Víctor Galán Grande
 */
public interface ImageGalleryListener extends BaseListListener<ImageEntry> {

    /**
     * Called when an item in the list is deleted.
     */
    void onImageEntryDeleted(long imageEntryId);

    /**
     * Called when an item is prepared for upload.
     */
    void onImageUploadStarted(Uri pictureUri, String title, String description, String changelog);

    /**
     * Called when an item is uploading.
     */
    void onImageUploadProgress(int totalBytes, int totalBytesSent);

    /**
     * Called when an item finishes uploading.
     */
    void onImageUploadEnd(ImageEntry entry);

    /**
     * Called when the View for uploading an image is instantiated.
     * The default behavior is to show the default view in a dialog.
     * To retain this behavior, all this method needs to do is return `false`.
     * To change the default behavior, use the {@link DefaultUploadDetailView#initializeUploadView}
     * method to initialize a custom View that extends {@link BaseDetailUploadView}.
     * Then return `true` to prevent the screenlet from executing the default behavior.
     *
     * @return boolean
     */
    boolean showUploadImageView(String actionName, Uri pictureUri, int screenletId);

    /**
     * Called when the screenlet provides the image upload View.
     * To inflate the default view, return `0` in this method.
     * Alternatively, display this View with a custom layout by returning its layout ID.
     * Such a layout must have {@link DefaultUploadDetailView} as its root class.
     *
     * @return layout ID or `0` to inflate the default view.
     */
    @LayoutRes
    int provideImageUploadDetailView();
}
